fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -750,6 +751,16 @@ public void execute() throws Exception {
                 coord.cancel(PPlanFragmentCancelReason.INTERNAL_ERROR, context.getState().getErrorMessage());
             }
 
+            if (parsedStmt instanceof InsertStmt) {
+                // Revert useMetadataCache to true after executing the insert query that
+                // insert into target_table select from hive_table.
+                InsertStmt insertStmt = (InsertStmt) parsedStmt;
+                List<Table> tables = new ArrayList<>();
+                AnalyzerUtils.collectSpecifyExternalTables(insertStmt.getQueryStatement(), tables, Table::isHiveTable);
```

Comment:
apply this flag in the https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/connector/RemoteFileOperations.java#L205. and set flag when building GetRemoteFilesParams in the https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnectorScanRangeSource.java#L131.